src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
```

Comment:
Actually it does: `StandardIntegrationFlow`.
But `IntegrationFlow` can't, because it is functional interface.
Although `IntegrationFlowBeanPostProcessor` overrides it to the `StandardIntegrationFlow`:
```
private Object processIntegrationFlowImpl(IntegrationFlow flow, String beanName) {
	IntegrationFlowBuilder flowBuilder = IntegrationFlows.from(beanName + ".input");
	flow.configure(flowBuilder);
	Object standardIntegrationFlow = processStandardIntegrationFlow(flowBuilder.get(), beanName);
	return isLambda(flow) ? standardIntegrationFlow : flow;
}
```
So, in most cases we are `SmartLifecycle`.

Only use-case when it isn't Lambda is direct interface implementation or `IntegrationFlowAdapter` extension.
In this case you have to implement `tLifecycle` yourself.

What am I missing ?